pkg/clusters/addons/loadimage/addon.go

```diff
@@ -0,0 +1,101 @@
+package loadimage
```

Comment:
Ultimately the goal of the cluster addons so far is to be usable on any `clusters.Cluster`, this addon is entirely tied to `kind` based clusters, breaking through the intended abstraction. It seems to me that we can move this same functionality as methods on the `kind` cluster type in `pkg/clusters/types/kind/` to isolate this to the cluster implementation that actually supports it, the effect of which is that you need to operate on it as it's underlying type (there's precedent for this in the addons). Let me know your thoughts?

Reply:
Which has the precedent?

Reply:
Sorry what I meant was that the addons do something similar to this, they each adhere to the interface but several of them have their own domain specific methods.